### PR TITLE
BREAKING CHANGE: Monaco version upgrade to 0.33.0

### PR DESCRIPTION
**Pull request by dileepyavan**
- [X] I have read the [Contributor Guide](https://github.com/nteract/nteract/blob/master/CONTRIBUTING.md)
- [X] I have updated the changelogs/current_changelog.md file with some information about the change that I am making the appropriate file.
- [X] I have validated or unit-tested the changes that I have made.
- [X] I have run through the TEST_PLAN.md to ensure that my change does not break anything else.

 [Monaco editor change log for 0.33.0 version](https://github.com/microsoft/monaco-editor/blob/main/CHANGELOG.md#0330)

### CODE REVIEW

**Review by barryt2 — APPROVED** [review]

**Comment by vivek1729** [comment]
@dileepyavan I just noticed that nteract also uses  monaco-editor in their jupyter extension application. Would you be able to update the webpack config for that project as well? Here's the [package.json](https://github.com/nteract/nteract/blob/main/applications/jupyter-extension/nteract_on_jupyter/package.json#L35) and [webpack config](https://github.com/nteract/nteract/blob/main/applications/jupyter-extension/nteract_on_jupyter/webpack.config.js#L12) for that project and I found it referencing an older version for the monaco-editor webpack plugin which won't be compatible with v 0.33.x

File: packages/monaco-editor/src/MonacoEditor.tsx [hunk 1]
```diff
@@ -215,7 +215,7 @@ export default class MonacoEditor extends React.Component<IMonacoProps> {
         fixedOverflowWidgets: true,
         find: {
           addExtraSpaceOnTop: false, // pops the editor out of alignment if turned on
-          seedSearchStringFromSelection: true, // default is true
+          seedSearchStringFromSelection: "always", // default is true
```
Comment (captainsafia) [inline packages/monaco-editor/src/MonacoEditor.tsx]:
```suggestion
          seedSearchStringFromSelection: "always", // default is "always"
```

File: packages/monaco-editor/src/theme.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export const customMonacoDarkTheme: monaco.editor.IStandaloneThemeData = {
   inherit: true,
   rules: [],
   colors: {
-    "editor.background": "#111"
+    "editor.background": "#111111"
```
Comment (captainsafia) [inline packages/monaco-editor/src/theme.ts:37]: Interesting -- is there a particular reason we can't leverage the hex code shorthand anymore?

**Review by captainsafia — COMMENTED** [review]
Is there a particular bug fix/feature we're looking to get from this major upgrade?

~Also, to capture the breaking change for conventional commits correctly, it'll have to be communicated via a `!` in the message. The "BREAKING CHANGE" is only applicable in the commit body AFAIK ([ref](https://www.conventionalcommits.org/en/v1.0.0/)).~

It looks like some commits have the right naming. I think that should be enough.

File: packages/monaco-editor/src/theme.ts (hunk 2 above)
Comment (dileepyavan) [reply packages/monaco-editor/src/theme.ts:37]: Monaco editor came up with new validation which checks for the length of the code and if its less than 6, throws invalid color code error.

**Comment by dileepyavan** [comment]
> 



> @dileepyavan I just noticed that nteract also uses monaco-editor in their jupyter extension application. Would you be able to update the webpack config for that project as well? Here's the [package.json](https://github.com/nteract/nteract/blob/main/applications/jupyter-extension/nteract_on_jupyter/package.json#L35) and [webpack config](https://github.com/nteract/nteract/blob/main/applications/jupyter-extension/nteract_on_jupyter/webpack.config.js#L12) for that project and I found it referencing an older version for the monaco-editor webpack plugin which won't be compatible with v 0.33.x

@vivek1729  Thanks for the review, I have updated the PR with mentioned changes.

**Comment by dileepyavan** [comment]
> Is there a particular bug fix/feature we're looking to get from this major upgrade?
> 
> ~Also, to capture the breaking change for conventional commits correctly, it'll have to be communicated via a `!` in the message. The "BREAKING CHANGE" is only applicable in the commit body AFAIK ([ref](https://www.conventionalcommits.org/en/v1.0.0/)).~
> 
> It looks like some commits have the right naming. I think that should be enough.

@captainsafia We requested monaco editor for a change regarding streaming hover results in case of multiple hover providers. This was added in https://github.com/microsoft/monaco-editor/blob/main/CHANGELOG.md#0310-10122021.

**Review by vivek1729 — APPROVED** [review]
Thanks Dileep! Your changes LGTM.

**Comment by vivek1729** [comment]
@captainsafia , I am merging the PR as the changes look complete to me. @dileepyavan can address any additional feedback you might have as a follow up PR.

**Pull request merged**